modules/google-maps/src/utils.js

```diff
@@ -0,0 +1,155 @@
+/* global document, google */
+import {Deck} from '@deck.gl/core';
+
+/**
+ * Get a new deck instance
+ * @param map (google.maps.Map) - The parent Map instance
+ * @param overlay (google.maps.OverlayView) - A maps Overlay instance
+ * @param [deck] (Deck) - a previously created instances
+ */
+export function createDeckInstance(map, overlay, deck) {
+  if (deck) {
+    if (deck.props.userData._googleMap === map) {
+      return deck;
+    }
+    // deck instance was created for a different map
+    destroyDeckInstance(deck);
+  }
+
+  const eventListeners = {
+    click: null,
+    mousemove: null,
+    mouseout: null
+  };
+
+  deck = new Deck({
+    canvas: createDeckCanvas(overlay),
+    initialViewState: {
+      longitude: 0,
+      latitude: 0,
+      zoom: 1
+    },
+    controller: false,
+    userData: {
+      _googleMap: map,
+      _eventListeners: eventListeners
+    }
+  });
+
+  // Register event listeners
+  for (const eventType in eventListeners) {
+    eventListeners[eventType] = map.addListener(eventType, evt =>
+      handleMouseEvent(deck, eventType, evt)
+    );
+  }
+
+  return deck;
+}
+
+function createDeckCanvas(overlay) {
+  const container = overlay.getPanes().overlayLayer;
+  const deckCanvas = document.createElement('canvas');
+  Object.assign(deckCanvas.style, {
+    // map container position is always non-static
+    position: 'absolute',
+    pointerEvents: 'none',
```

Comment:
minor: `pointerEvents: 'none'` shouldn't be necessary anymore – mouse events do not reach overlayLayer` anyway.